components/service/glean/src/test/java/mozilla/components/service/glean/scheduler/MetricsPingSchedulerTest.kt

```diff
@@ -426,13 +426,15 @@ class MetricsPingSchedulerTest {
 
     @Test
     fun `schedulePingCollection must correctly append a work request to the WorkManager`() {
-        val mps = MetricsPingScheduler(ApplicationProvider.getApplicationContext<Context>())
+        // Replacing the singleton's metricsPingScheduler here since doWork() refers to it when
+        // the worker runs, otherwise we can get a lateinit property is not initialized error.
+        Glean.metricsPingScheduler = MetricsPingScheduler(ApplicationProvider.getApplicationContext<Context>())
```

Comment:
Good catch. Is this the only test that requires it?

Reply:
I believe so, every other call to `schedulePingCollection` is really just using `verify` to make sure it gets called or doesn't get called.  This is the only one where we get into WorkManager to ensure it's there, which is what was enqueuing the worker and causing the issue.